sourced/ml/algorithms/id_splitter/pipeline.py

```diff
@@ -0,0 +1,274 @@
+import argparse
+from datetime import datetime
+import logging
+import pickle
+import random
+import os
+from typing import Callable, Iterable, Tuple
+import warnings
+
+import numpy
+from keras import backend as kbackend
+from keras.callbacks import CSVLogger, TensorBoard, ModelCheckpoint, LearningRateScheduler
+try:
+    import tensorflow as tf
+except ImportError:
+    warnings.warn("Tensorflow is not installed, dependent functionality is unavailable.")
+
+from sourced.ml.algorithms.id_splitter import prepare_features
+
+
+EPSILON = 10 ** -8
+DEFAULT_THRESHOLD = 0.5  # threshold that is used to binarize predictions of the model.
+
+
+def set_random_seed(seed: int) -> None:
+    """
+    Fixes random seed for reproducibility.
+
+    :param seed: seed value.
+    """
+    numpy.random.seed(seed)
+    random.seed(seed)
+    tf.set_random_seed(seed)
+
+
+def to_binary(matrix: numpy.array, threshold: float, inplace: bool=True) -> numpy.array:
+    """
+    Helper function to binarize a matrix.
+
+    :param matrix: matrix as a numpy.array.
+    :param threshold: if value >= threshold then the value will be 1, else 0.
+    :param inplace: whether to modify the matrix inplace or not.
+    :return: the binarized matrix.
+    """
+    mask = matrix >= threshold
+    if inplace:
+        matrix_ = matrix
+    else:
+        matrix_ = matrix.copy()
+    matrix_[mask] = 1
+    matrix_[numpy.logical_not(mask)] = 0
+    return matrix_
+
+
+def precision_np(y_true: numpy.array, y_pred: numpy.array, epsilon: float=EPSILON) -> float:
+    """
+    Computes the precision metric, a metric for multi-label classification of
+    how many selected items are relevant.
+
+    :param y_true: ground truth labels - expect binary values.
+    :param y_pred: predicted labels - expect binary values.
+    :param epsilon: to avoid division by zero.
+    :return: precision metric.
+    """
+    true_positives = np.sum(y_true * y_pred)
+    predicted_positives = np.sum(y_pred)
```

Comment:
`np` instead of `numpy`

Is this function needed at all?

Reply:
@EgorBu  introduced the numpy metrics to calculate the results during evaluation while the tensor metrics are used during the training to show the progress. I kept them.